reference/enterprise-api-ref/inventory.markdown

```diff
@@ -0,0 +1,211 @@
+
+**Example Request Body:**
+
+
+```
+{  
+   "sort":"Host name",
+   "filter":{  
+      "Host name":{  
+         "l":"%value1%",
+         "nil":"%value2%"
+      }
+   },
+   "select":[  
+      "Host name",
+      "OS type",
+      "IPv4 addresses",
+      "CFEngine version",
+      "Ports listening"
+   ],
+   "hostContextExclude":[  
+      "policy_server"
+   ]
+}
+```
+
+**Example response:**
+
+
+```
+{
+  "data": [
+    {
+      "header": [
+        {
+          "columnName": "Host name",
+          "columnType": "STRING"
+        },
+        {
+          "columnName": "OS type",
+          "columnType": "STRING"
+        },
+        {
+          "columnName": "IPv4 addresses",
+          "columnType": "STRING"
+        },
+        {
+          "columnName": "CFEngine version",
+          "columnType": "STRING"
+        },
+        {
+          "columnName": "Ports listening",
+          "columnType": "STRING"
+        }
+      ],
+      "queryTimeMs": 30,
+      "rowCount": 2,
+      "rows": [
+        [
+          "host1.cfengine.com",
+          "linux",
+          "128.30.23.33",
+          "3.10.0",
+          "22, 25, 80, 443, 5308, 9000"
+        ],
+        [
+          "host2.cfengine.com",
+          "linux",
+          "184.45.1.75",
+          "3.10.0",
+          null
+        ]
+      ]
+    }
+  ],
+  "meta": {
+    "count": 1,
+    "page": 1,
+    "timestamp": 1496222472,
+    "total": 1
+  }
+}
+```
+
+## List of variables
+
+**URI:** https://hub.cfengine.com/api/variables-dictionary
+
+**Method:** GET
+
+Shows list of all inventory variables available in the system. 
+
+See more details: 
+* [Custom Inventory][Custom Inventory]
+
+
+**Example response:**
+
+
+```
+[
+  {
+    "id": 2,
+    "attribute_name": "BIOS vendor",
+    "category": "Hardware",
+    "readonly": 1,
+    "type": "string",
+    "convert_function": null
+  },
+  {
+    "id": 3,
+    "attribute_name": "BIOS version",
+    "category": "Hardware",
+    "readonly": 1,
+    "type": "string",
+    "convert_function": null
+  }
+]
+```
+
+## Edit variable
+
+**URI:** https://hub.cfengine.com/api/variables-dictionary/:id
+
+**Method:** PATCH
+
+Set variables type (int/string..). This is needed for applying filtering in correct format.
+Only `readonly - 0` variable can be edited
+
+**Parameters:**
+
+* **id** *(integer)*
+    Variable Id
+* **category** *(string)*
+    Category of variable
+* **type** *(string)*
+    Variable's type. Allowed values: int, real, slist, string
+* **convert_function** *(string)*
+    Convert Function. 
+    Emp.: `cf_clearSlist` - to transform string like `{"1", "2"}` to `1, 2`
+
+**Example Request Body:**
+
+```
+{
+ "category":"Hardware", 
+ "type": "int"
+}
+```
+
+**Example response:**
+
+```
+{
+  "id": 1,
+  "attribute_name": "Architecture",
+  "category": "Hardware",
+  "readonly": 0,
+  "type": "slist",
+  "convert_function": "cf_clearSlist"
+}
+```
+
+## Add custom variables to variables dictionary 
```

Comment:
I don't understand what this is for. Is this to create a new category like Hardware or Software but user defined? Or is it for mapping a variable into an existing group?